plugins/obs-outputs/mp4-output.c

```diff
@@ -250,17 +252,22 @@ static bool mp4_output_start(void *data)
 
 	os_atomic_set_bool(&out->stopping, false);
 
-	/* get path */
 	obs_data_t *settings = obs_output_get_settings(out->output);
-	const char *path = obs_data_get_string(settings, "path");
-	dstr_copy(&out->path, path);
-
 	out->max_time = obs_data_get_int(settings, "max_time_sec") * 1000000LL;
 	out->max_size = obs_data_get_int(settings, "max_size_mb") * 1024 * 1024;
 	out->split_file_enabled = obs_data_get_bool(settings, "split_file");
 	out->allow_overwrite = obs_data_get_bool(settings, "allow_overwrite");
 	out->cur_size = 0;
 
+	/* Get path */
+	const char *path = obs_data_get_string(settings, "path");
+	if (path && *path) {
```

Comment:
I suppose this could be replaced with `if (obs_data_has_user_value(settings, "path")) {` but it's probably fine either way.

Reply:
We need to fetch the `path` string in *most* cases anyway, so I think it's fair to keep.